capycli/bom/create_components.py

```diff
@@ -366,6 +368,10 @@ def upload_file(
             filename = str(CycloneDxSupport.get_ext_ref_binary_file(cx_comp))
             filehash = str(CycloneDxSupport.get_binary_file_hash(cx_comp))
 
+        if filename is not None and filename.endswith('.git'):
+            print_red("    WARNING: resetting filename to prevent uploading .git file")
+            filename = None
```

Comment:
I'm not completely against this, but do we really need this extra check? There are plenty more of extensions we probably don't want to upload like .exe, .scr, .pif, .so, ... Is there an important reason to hard-code this extension? Probably someone one day has a valid use case to upload "snapshot.git"?

Reply:
Hi @gernot-h,
all of these are valid questions and @nzupan should provide answers.  
What I know: sometimes the source code detection or source code upload do not work as inteded.  
The result is an attachment on SW360 that cannot be used, because it does not contain source code.  
My assumption was that thesecode changes mitigate this problem.

Reply:
That was exactly the intention behind the change. What I've observed in practice is that on many javascript packages, the .git is being uploaded instead of the source code archive file due to the source code detection not working as intended or the package's metadata being wrong.
Due to the JS projects often containing many dependencies, preventing uploading .git files can prevent a huge amount of manual work or in the worst case wrong uploads/moderation requests (if not enough care is put in by the dev teams when polishing SBOM files).
@gernot-h , but what you pointed out is a valid use case. I believe that a good way to improve this check is to apply it explicitly only for the SOURCE AND BINARY file type.
If you both agree, I would open another merge request for it.